src/agentPlot/display_graph.py

```diff
@@ -0,0 +1,130 @@
+
+        for num, edge in enumerate(graph.network.edges):
+            if tuple(sorted((get_id(s.get('sender')) - 1, (get_id(s.get('receiver')) - 1)))) == edge:
+                msg = s.get('content')
+                edge_labels[edge] = "{}/{}".format(np.around(msg.money, 2), np.around(msg.resource, 2))
+
+                if msg.type == OfferType.INITIAL_OFFER:
+                    edge_colors[num] = "blue"
+                elif msg.type == OfferType.COUNTER_OFFER:
+                    edge_colors[num] = "yellow"
+                elif msg.type == OfferType.ACCEPTING_OFFER:
+                    edge_colors[num] = "green"
+                elif msg.type == OfferType.BREAKDOWN_OFFER:
+                    edge_colors[num] = "red"
+                elif msg.type == OfferType.CONFIRMATION_OFFER:
+                    edge_colors[num] = "green"
+                else:
+                    edge_colors[num] = "black"
+
+    pos = nx.kamada_kawai_layout(graph.network)
+
+    nx.draw_networkx_edges(graph.network, pos, edge_color=edge_colors)
+    nx.draw_networkx_edge_labels(graph.network, pos, with_labels=True, edge_labels=edge_labels)
+    nx.draw_networkx_nodes(graph.network, pos)
+    nx.draw_networkx_labels(graph.network, pos, labels=node_labels)
+
+
+def real_time_plot(graph, handlers):
+    """
+    Plots simulation graph in real time
+    :param graph: agent_net graph
+    :param handlers: dictionary containing logs from the system
+    """
+    global edge_colors
+    edge_colors = ["black" for i in range(len(graph.network.edges))]
+    global edge_labels
+    edge_labels ={}
+    for edge in graph.network.edges:
+        edge_labels[edge] = '-/-'
+    global node_labels
+    node_labels = {}
+    for node in graph.network.nodes:
+        node_labels[node] = '-/-'
+    plt.ion()
+    plt.show()
+
+    message = handlers.get(logger.EVENT_MESSAGE_SENT)
+    status = handlers.get(logger.EVENT_AGENT_STATE_CHANGED)
+
+    if message is not None:
+        for s in message:
+            for num, edge in enumerate(graph.network.edges):
+                if tuple(sorted((get_id(s.get('sender')) - 1, (get_id(s.get('receiver')) - 1)))) == edge:
+                    msg = s.get('content')
+                    edge_labels[edge] = "{}/{}".format(np.around(msg.money, 2), np.around(msg.resource, 2))
+
+
+                    if msg.type == OfferType.INITIAL_OFFER:
+                        edge_colors[num] = "blue"
+                    elif msg.type == OfferType.COUNTER_OFFER:
+                        edge_colors[num] = "yellow"
+                    elif msg.type == OfferType.ACCEPTING_OFFER:
+                        edge_colors[num] = "green"
+                    elif msg.type == OfferType.BREAKDOWN_OFFER:
+                        edge_colors[num] = "red"
+                    elif msg.type == OfferType.CONFIRMATION_OFFER:
+                        edge_colors[num] = "green"
+                    else:
+                        edge_colors[num] = "black"
```

Comment:
można by było zrobić handler który to ustawia (edge_colors itp) zamiast trzymać dicta